examples/signature-agent-card-and-registry/src/lib.rs

```diff
@@ -0,0 +1,159 @@
+use base64::{Engine as _, engine::general_purpose};
+use ed25519_dalek::SigningKey;
+use indexmap::map::IndexMap;
+use rand_chacha::rand_core::{RngCore, SeedableRng};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+use time::Duration;
+use web_bot_auth::{
+    components::{CoveredComponent, DerivedComponent, HTTPField, HTTPFieldParametersSet},
+    keyring::{Algorithm, Thumbprintable},
+    message_signatures::{MessageSigner, UnsignedMessage},
+};
+use worker::*;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
+struct SignatureAgentCard {
+    client_name: String,
+    contacts: Vec<String>,
+    keys: Vec<Thumbprintable>,
+}
+
+struct SignatureHeaderGenerator<'a> {
+    req: &'a HttpRequest,
+    digest_header: String,
+    outputs: (String, String),
+}
+
+impl UnsignedMessage for SignatureHeaderGenerator<'_> {
+    fn fetch_components_to_cover(
+        &self,
+    ) -> IndexMap<web_bot_auth::components::CoveredComponent, String> {
+        IndexMap::from_iter([
+            (
+                CoveredComponent::Derived(DerivedComponent::Authority { req: true }),
+                self.req.uri().host().unwrap().to_string(),
+            ),
+            (
+                CoveredComponent::HTTP(HTTPField {
+                    name: "content-digest".to_string(),
+                    parameters: HTTPFieldParametersSet(vec![]),
+                }),
+                self.digest_header.clone(),
+            ),
+        ])
+    }
+
+    fn register_header_contents(&mut self, signature_input: String, signature_header: String) {
+        self.outputs = (
+            format!("sig1={}", signature_input),
+            format!("sig1={}", signature_header),
+        )
+    }
+}
+
+#[event(fetch)]
+async fn fetch(req: HttpRequest, env: Env, _ctx: Context) -> Result<Response> {
+    let kv = env.kv("signed-agent-registry-hostnames")?;
+    let host = req.uri().host().ok_or(worker::Error::RouteNoDataError)?;
+
+    match req.uri().path() {
+        "/" => {
+            let list = kv.list().limit(1000).execute().await?;
+            Response::ok(
+                list.keys
+                    .into_iter()
+                    .map(|key| {
+                        format!(
+                            "https://{}/.well-known/http-message-signatures-directory",
```

Comment:
use `http` when on localhost, or correct the readme to include [mkcert](https://github.com/FiloSottile/mkcert)-like tool